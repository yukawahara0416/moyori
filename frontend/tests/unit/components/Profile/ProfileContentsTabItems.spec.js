import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Profile/ProfileContentsTabItems.vue'

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    tabs: 'tab-1',
    user: { data: { id: 1 }, posts: [{ data: { id: 1 } }] }
  }

  wrapper = shallowMount(Component, {
    propsData
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('props', () => {
  it('tabs', () => {
    expect(wrapper.props().tabs).toEqual(propsData.tabs)
  })
  it('user', () => {
    expect(wrapper.props().user).toEqual(propsData.user)
  })
})

// describe('computed', () => {
//   it('childTabs', () => {})
// })

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
