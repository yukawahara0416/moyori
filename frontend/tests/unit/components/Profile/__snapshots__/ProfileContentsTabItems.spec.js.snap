// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-tabs-items-stub
  activeclass="v-window-item--active"
  nexticon="$next"
  previcon="$prev"
  value="tab-1"
>
  <v-tab-item-stub
    value="tab-1"
  >
    <card-container-stub
      spots="[object Object]"
      type="posts"
    />
  </v-tab-item-stub>
   
  <v-tab-item-stub
    value="tab-2"
  >
    <card-container-stub
      spots=""
      type="wifi_withs"
    />
  </v-tab-item-stub>
   
  <v-tab-item-stub
    value="tab-3"
  >
    <card-container-stub
      spots=""
      type="comments"
    />
  </v-tab-item-stub>
   
  <v-tab-item-stub
    value="tab-4"
  >
    <card-container-stub
      spots=""
      type="likes"
    />
  </v-tab-item-stub>
</v-tabs-items-stub>
`;
